batch/batch/worker/worker.py

```diff
@@ -2231,8 +2231,9 @@ async def cleanup(self):
                     except asyncio.CancelledError:
                         raise
                     except Exception:
-                        log.exception(f'error while unmounting cloudfuse for {self.jvm_name} for job {self.id}')
-                        raise IncompleteJVMCleanupError
+                        raise IncompleteJVMCleanupError(
+                            f'while unmounting fuse blob storage {bucket} from {mount_path} for {self.jvm_name} for job {self.id}'
+                        )
```

Comment:
Name the exception and add a `from e` so that we get the fulls tack trace in the log.exception later.